tests/automated_validation/data_transformation/test_data_schema.py

```diff
@@ -0,0 +1,110 @@
+import pandas as pd
+import pytest
+from pandera.errors import SchemaError
+
+from vivarium_testing_utils.automated_validation.data_transformation.data_schema import (
+    DrawData,
+    RatioData,
+    SimOutputData,
+    SingleNumericValue,
+)
+
+
+def test_single_numeric_value() -> None:
+    """Test that the SingleNumericValue schema correctly validates a DataFrame with a single numeric column."""
+    schema = SingleNumericValue
+    data = pd.DataFrame({"value": [1, 2, 3]}, index=pd.Index([0, 1, 2], name="index"))
+    schema.validate(data)
```

Comment:
How come these tests call validate but nowhere else? When exactly do the things get validated? I assume it's related to the `pa.check_type` decorators?

Maybe it's just worth having a quick discussion explaining how/when these schema checks are actually triggered.